packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -155,35 +165,58 @@ export class NftDetectionController extends StaticIntervalPollingControllerV1<
 
   private getOwnerNftApi({
     address,
-    offset,
+    next,
   }: {
     address: string;
-    offset: number;
+    next?: string;
   }) {
-    return `${OPENSEA_PROXY_URL}/assets?owner=${address}&offset=${offset}&limit=50`;
+    return `${OPENSEA_PROXY_URL}/chain/${
+      OpenSeaV2ChainIds.ethereum
+    }/account/${address}/nfts?limit=200&next=${next ?? ''}`;
   }
 
   private async getOwnerNfts(address: string) {
-    let nftApiResponse: { assets: ApiNft[] };
+    let nftApiResponse: OpenSeaV2ListNftsResponse;
     let nfts: ApiNft[] = [];
-    let offset = 0;
-    let pagingFinish = false;
-    /* istanbul ignore if */
+    let next;
+
     do {
       nftApiResponse = await fetchWithErrorHandling({
-        url: this.getOwnerNftApi({ address, offset }),
+        url: this.getOwnerNftApi({ address, next }),
         timeout: 15000,
       });
 
       if (!nftApiResponse) {
         return nfts;
       }
 
-      nftApiResponse?.assets?.length !== 0
-        ? (nfts = [...nfts, ...nftApiResponse.assets])
-        : (pagingFinish = true);
-      offset += 50;
-    } while (!pagingFinish);
+      const newNfts = await Promise.all(
+        nftApiResponse.nfts.map(async (nftV2) => {
+          const nftV1 = mapOpenSeaNftV2ToV1(nftV2);
+
+          // If the image hasn't been processed into OpenSea's CDN, the image_url will be null.
+          // Try fetching the NFT individually, which returns the original image url from metadata if available.
+          if (!nftV1.image_url && nftV2.metadata_url) {
+            const nftDetails: OpenSeaV2GetNftResponse | undefined =
+              await safelyExecute(() =>
+                timeoutFetch(
+                  this.getNftApi({
```

Comment:
Hmm, this could mean a lot more network requests.

I wonder if we could cache these somehow to prevent this.

Reply:
Yeah.  Although it's not the common case.  Saw no occurrences in most wallets, and only on a few tokens when it does occur.  I could tally more stats on the %.  Also the `&& metadata_url` avoids doing it some cases where it wouldn't return anything useful.

Reply:
Good point! OK, maybe we highlight it to the mobile team as a risk, and we can add caching later if necessary

Reply:
Was testing occurrences with a script like this:

```
let next = '', account = '';
do {
  const nfts = await(
    await fetch(
      `https://api.opensea.io/api/v2/chain/ethereum/account/${account}/nfts?limit=200&next=${next}`,
      { headers: { 'x-api-key': '' } }
    )).json();

    for (nft of nfts.nfts) {
    if (!nft.image_url && nft.metadata_url) {
      console.log(nft);
    }
  }
} while (next = nfts.next);
```